_posts/2016-03-31-Adding-TypeScript-to-Ionic-Framework.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: post
+author: jan_de_wilde
+title: 'Adding TypeScript to Ionic Framework'
+image: /img/ionic-and-typescript.jpg
+tags: [TypeScript,Ionic]
+category: Ionic
+comments: true
+---
+
+### Ionic and TypeScript sitting in a tree
+
+So, [TypeScript](http://www.typescriptlang.org/) is the all new thing that allows you to use features from ES6 (or ES2015), ES7 and beyond. Say bye to loosely typed variables and say hello to modules, classes, interfaces and so much more.
+
+In order to use TypeScript in an Ionic Framework project there are a few small things you need to do to get things running.
+
+## 1. Install and configure the gulp package
+
+* Install the **gulp-tsc package** and save it to the development dependencies in `package.json`
+
+   {% highlight text %}npm install gulp-tsc --save-dev{% endhighlight %}
+
+* Next, require the package in your **gulpfile.js** like so
+
+   {% highlight text %}var typescript = require(‘gulp-tsc’);{% endhighlight %}
+
+* Add the following line to the paths object: `ts: ['./src/*.ts', './src/**/*.ts']`. You may have noticed two things here: I’m using a **src folder** to place all my TypeScript files in which means I'm **not using the www folder** that Ionic provides by default. Next to that I’m also targeting subfolders in that folder because I’m bundling my logic based on AngularJS modules. You can read more about structuring an AngularJS project in the [John Papa AngularJS Style Guide](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#application-structure-lift-principle).
+
+* Add our compile task
+
+   {% highlight text %}
+gulp.task('compile', function(){
+    gulp.src(paths.ts)
+        .pipe(typescript({ emitError: false }))
+        .pipe(gulp.dest('www/‘));
+});{% endhighlight %}
+
+* Add our task to watch
+
+   {% highlight text %}
+gulp.task('watch', function () {
+    gulp.watch(paths.sass, ['sass’]);
+    gulp.watch(paths.ts, [‘compile']);
+});{% endhighlight %}
+
+* Now change the **ionic.project** file and add the compile task to the gulpStartupTasks. If the gulpStartupTasks section is not present at all, just add it anyway.
+
+   {% highlight text %}
+"gulpStartupTasks": [
+    "sass",
+    "compile",
+    "watch"
+]{% endhighlight %}
+
+## 2. Add TSD
+
+TSD is a **TypeScript Definition manager** for [DefinitelyTyped](http://definitelytyped.org/). TypeScript used TypeScript Definition files so it knows how to handle the TypeScript you are writing and gives you intellisense. Let's install TSD so we can continue.
```

Comment:
Dit weet ik niet goed, maar is het ook niet handig om de TSD als dev dependency te installen? of is dit niet nodig?